### ISSUE TITLE
`star_utils:get_phot_info` should raise non-zero `ierr` if L < 0

### ISSUE THREAD

**Issue by warrickball** [MESAHub/mesa#280]
The various code paths in `star_utils:get_phot_info` currently avoid dealing with a negative luminosity (which causes problems in e.g. `set_phot_info` when trying to evaluate Teff ~ (L/4πR²σ)⁰·²⁵) by limiting the output luminosity to `max(1, L)`. As @adamjermyn suggested in [PR 278](https://github.com/MESAHub/mesa/pull/278), we should probably rather have the various routines raise a non-zero `ierr` rather than make up a value of `L` and pretend everything is OK.

**Comment by adamjermyn** [comment]
I suppose the counterargument would be that maybe the solver learns something by evaluating with a small but non-negative L, and we'd be cutting off that iteration prematurely by raising an error. I don't find that argument terribly convincing but it is at least in principle an empirical case someone could make...

**Comment by warrickball** [comment]
Yeah, I figured step one here would be to *try* raising `ierr /= 0` and see what happens.

**Comment by warrickball** [comment]
If so, where's the "photosphere"?

**Comment by adamjermyn** [comment]
I don't think photospheres make sense in that case...

**Comment by adamjermyn** [comment]
Like in that case the radiation field doesn't become strictly outward-facing at infinity, and I think you need that to define a photosphere?

**Comment by warrickball** [comment]
That's what I'm thinking too. But in that case you might not want to call a subroutine who's name implies you want information at the photosphere. 

That said, I'm not sure that isn't what we're doing anyway when using the `'irradiated'` atmosphere option. I just had a quick look in `atm` and it looks like even that assumes a positive luminosity, since it computes some sort of temperature using the Stefan–Boltzman law. So all this possibly deserves a rethink but at a higher level than `get_phot_info`.

**Comment by adamjermyn** [comment]
Agreed. I don't think get_phot_info is the right place for this change.

-Adam
